var/spack/repos/builtin/packages/adios2/package.py

```diff
@@ -281,3 +283,50 @@ def setup_run_environment(self, env):
             env.prepend_path("HDF5_PLUGIN_PATH", os.path.dirname(all_libs[idx]))
         except ValueError:
             pass
+
+    @run_after("install")
+    def setup_install_tests(self):
+        """
+        Copy the example files after the package is installed to an
+        install test subdirectory for use during `spack test run`.
+        """
+        extra_install_tests = [join_path("testing", "install", "C")]
+        self.cache_extra_test_sources(extra_install_tests)
+
+    def test_run_executables(self):
+        """Run installed adios2 executables"""
+
+        commands_and_args = [("bpls", ["-v", "-V"])]
+
+        for cmd, opts in commands_and_args:
+            with test_part(
+                self, f"test_adios2_cmds_{cmd}", purpose=f"run installed adios2 executable {cmd}"
+            ):
+                exe = which(join_path(self.prefix.bin, cmd))
+                exe(*opts)
+
+    def test_build_example(self):
+        """Build and run an example program"""
+        src_dir = self.test_suite.current_test_cache_dir.testing.install.C
+        test_stage_dir = self.test_suite.test_dir_for_spec(self.spec)
+
+        # Create the build tree within this spec's test stage dir so it gets
+        # cleaned up automatically
+        build_dir = tempfile.mkdtemp(dir=test_stage_dir)
+
+        std_cmake_args = []
+
+        if "+mpi" in self.spec:
+            mpi_exec = join_path(self.spec["mpi"].prefix, "bin", "mpiexec")
+            std_cmake_args.append("-DMPIEXEC_EXECUTABLE={0}".format(mpi_exec))
```

Comment:
Nit.  You're using f-strings elsewhere, so why not here:

```suggestion
            std_cmake_args.append(f"-DMPIEXEC_EXECUTABLE={mpi_exec}")
```

Reply:
If I used f-strings elsewhere, it was by accident, I was still under the impression we had to use `format()` in spack.  I'll make it consistent.

Reply:
Last I heard we're switching to f-strings in the core so I've been recommending their use in packages.  There was at least one exception but all I remember is it was a special case where maybe the f-string argument wasn't straightforward.

Where did you hear we should be using

Reply:
I didn't hear that anywhere.  At one point, probably a long time ago, not all python versions supported by spack had f-string support, and likely my brain was just stuck there.

Reply:
Okay.  Thought maybe I missed something.